packages/schematics/src/command-line/shared.ts

```diff
@@ -106,6 +110,25 @@ function parseGitOutput(command: string): string[] {
     .filter(a => a.length > 0);
 }
 
+export function getImplicitDependencies(
+  projects: ProjectNode[]
+): ImplicitDependencies {
+  const nxJson = readNxJson();
+
+  if (!nxJson.implicitDependencies) {
+    return {};
+  }
+
+  Object.entries<'*' | string[]>(nxJson.implicitDependencies).forEach(
+    ([key, value]) => {
+      if (value === '*') {
```

Comment:
implicitdependencies are user input, so it can be invalid. We should protect our users against typos.

If you look at `getProjectNodes`, you will see that we check that nx.json and angular.json are in sync. We need to extract that logic into a separate function and extend that logic. We need to check:

* projects in angular.json and projects in nx.json have the same keys.
* implicitDependencies point to real projects.
 * projects in angular.json located in libs are in sync with the paths property tsconfig.json 

That function should throw if one of these doesn't hold.

We then should invoke this function in getProjectNodes and in getImplicitDependencies.

Reply:
I've held off on the last one for now.
> * projects in angular.json located in libs are in sync with the paths property tsconfig.json

* If we use the `angular.json` as truth:
  * There might be `libs` that don't need to be in `tsconfig.json`. Maybe like `assets` or `styles`?
    * I don't want to break `affected` for those projects.. 
    * Typescript compilation will already break if its not defined
      * The issue is that users might not know how exactly to resolve it...
  * If we made this a tslint rule it would be more precise.
    * If an import points to where we think a lib should be but isn't resolvable we could autofix it?
      * Is this possible? 
    * Would it be overwritten by the compiler error?